autosklearn/evaluation/train_evaluator.py

```diff
@@ -134,18 +140,25 @@ def _fit_with_budget(
     ):
         if model.estimator_supports_iterative_fit():
             budget_factor = model.get_max_iter()
-            Xt, fit_params = model.fit_transformer(X_train[train_indices],
-                                                   Y_train[train_indices])
+            Xt, fit_params = model.fit_transformer(
```

Comment:
I'm wondering whether you need to update all the `X_train` and `y_train` in this file to allow for sparse and numpy, too?